src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationKeyComparer.cs

```diff
@@ -29,49 +29,64 @@ public class ConfigurationKeyComparer : IComparer<string>
         /// <returns>Less than 0 if x is less than y, 0 if x is equal to y and greater than 0 if x is greater than y.</returns>
         public int Compare(string? x, string? y)
         {
-            string[] xParts = x?.Split(_keyDelimiterArray, StringSplitOptions.RemoveEmptyEntries) ?? Array.Empty<string>();
-            string[] yParts = y?.Split(_keyDelimiterArray, StringSplitOptions.RemoveEmptyEntries) ?? Array.Empty<string>();
-
-            // Compare each part until we get two parts that are not equal
-            for (int i = 0; i < Math.Min(xParts.Length, yParts.Length); i++)
+            if (x is not null && y is not null)
             {
-                x = xParts[i];
-                y = yParts[i];
+                if (x!.Contains(ConfigurationPath.KeyDelimiter) && y!.Contains(ConfigurationPath.KeyDelimiter))
```

Comment:
```suggestion
                if (x.Contains(ConfigurationPath.KeyDelimiter) && y!.Contains(ConfigurationPath.KeyDelimiter))
```
Shouldn't be needed here (and below), as above is the check for `x is not null`.